discord_connector.py

```diff
@@ -158,128 +162,183 @@ def _handle_test_connectivity(self, param):
+            "cef": {
+                "URL": "",
+                "type": "",
+                "Description": ""
+            }
+        }
+
+        self.save_progress("working on embeds")
+        for embed in message.embeds:
+            self.save_progress("embed: {}".format(embed.to_dict))
+            embeds.append(self.create_embed_artifact(embed, artifact))
+
+        self.save_progress("working on attachments")
+        for attachment in message.attachments:
+            self.save_progress("attachment: {}".format(attachment.to_dict()))
+            attachments.append(self.create_attachment_artifact(attachment, artifact))
+
+        return attachments, embeds
+
+    # convert to use strategy pattern???
+
+    def create_embed_artifact(self, embed, artifact):
+        artifact["name"] = f"embed: {embed.title}"
+        artifact["cef"]["URL"] = embed.url
+        artifact["cef"]["Description"] = embed.description
+        status, creation_message, artifact_id = BaseConnector.save_artifact(self, artifact)
+        return artifact_id
+
+    def create_attachment_artifact(self, attachment, artifact):
+        artifact["name"] = f"attachment: {attachment.filename}"
+        artifact["cef"]["URL"] = attachment.url
+        artifact["cef"]["Description"] = attachment.description
+        artifact["cef"]["Type"] = attachment.content_type
+        status, creation_message, artifact_id = BaseConnector.save_artifact(self, artifact)
+        return artifact_id
+
+    def parse_message(self, message, attachments, embeds):
+
+        # not sure if those ifs are necessary or if those are not breaking return types
+        return {
+            "message origin": {
+                "channel id": message.channel.id,
+                "channel name": message.channel.name,
+            },
+            "message data": {
+                "created at": str(message.created_at),
+                "edited at": str(message.edited_at) if message.edited_at is not None else "message was not edited",
+            },
+            "author data": {
+                "author id": message.author.id,
+                "author name": message.author.name,
+            },
+            "jump url": message.jump_url,
+            "flags": self.parse_message_flags(message) if [] else "no flags",
+            "attachments": attachments if attachments else "no attachments",
+            "embeds": embeds if embeds else "no embeds",
+            "content": message.content
+        }
+
+    def parse_message_flags(self, message):
+        true_flags = []
+
+        # is there a better way than iterating all flags?
+        # dir() returns too much
+        for flag in ['crossposted', 'ephemeral', 'failed_to_mention_some_roles_in_thread', 'has_thread',
+                     'is_crossposted', 'loading', 'silent', 'source_message_deleted', 'suppress_embeds',
+                     'suppress_notifications', 'urgent', 'voice']:
+
+            if getattr(message.flags, flag) is True:
```

Comment:
Aren't they always of `bool` type? If you only want to get those of `True` value then `is True` is redundant here.